### PR TITLE
Ci

### PR DESCRIPTION
**Pull request by Guillemdb**
Making PR according to @vmarkovtsev suggestions. 

> Add CI. The repo is empty so it passes. CI should additionally test packaging btw.

I don't know how to test packaging. Can you send me a tutorial or something? I was planning to do it after the first PR was merged.

### CODE REVIEW

**Comment by Guillemdb** [comment]
The test that fails is because some changes were made in Modelforge and it needs to be fxed.

**Comment by zurk** [comment]
`.ci` directory should be added to `.gitignore`
Makefile pulls it automatically.

**Comment by zurk** [comment]
can you squash it into one commit to avoid having '`ci' in git history?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+language: python
+sudo: true
+dist: xenial
+git:
```
Comment (vmarkovtsev) [inline .travis.yml]: The depth is not required here

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+FROM ubuntu:18.04
+
+ENV BROWSER=/browser \
+    LC_ALL=en_US.UTF-8
+
+COPY requirements.txt ml_core/requirements.txt
+
+RUN apt-get update && \
+    apt-get install -y --no-install-suggests --no-install-recommends \
+      ca-certificates locales libxml2 libxml2-dev gcc g++ wget \
```
Comment (vmarkovtsev) [inline Dockerfile]: Can you please try without `libxml2 libxml2-dev` - the new bblfsh client is supposed to not need it

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Cython>=0.28,<1.0; python_version == '3.7'
+PyStemmer==1.3.0
+bblfsh==2.12.7
+modelforge==0.12.1
+numpy==1.16.2
+humanize==0.5.1
```
Comment (vmarkovtsev) [inline requirements.txt]: humanize is probably included into modelforge

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Cython>=0.28,<1.0; python_version == '3.7'
+PyStemmer==1.3.0
+bblfsh==2.12.7
```
Comment (vmarkovtsev) [inline requirements.txt]: Let's require bblfsh v3

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+modelforge==0.12.1
+numpy==1.16.2
+humanize==0.5.1
+pygments==2.3.1
+keras==2.2.4
+scikit-learn==0.20.3
+tqdm==4.31.1
```
Comment (vmarkovtsev) [inline requirements.txt]: modelforge should require it already (but please check)

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+exclude_packages = (
+    ("sourced.ml.core.tests", "sourced.ml.core.tests.source")
+    if not os.getenv("ML_CORE_SETUP_INCLUDE_TESTS", False)
+    else ()
+)
+
+
+setup(
+    name="sourced-ml-core",
+    description="Framework for machine learning on source code. "
```
Comment (vmarkovtsev) [inline setup.py]: We should change this: this is not a framework anymore

Comment (Guillemdb) [reply setup.py]: I changed it to
>Library containing the core algorithms for machine learning on source code.

File: requirements.txt (hunk 4 above)
Comment (Guillemdb) [reply requirements.txt]: I'm assuming you use semantic versioning and asked for bblfh >= 3.0.0

Comment (m09) [reply requirements.txt]: given the state of bblfsh 3 it will probably require a separate PR (probably not small)

Comment (vmarkovtsev) [reply requirements.txt]: Fair enough

Comment (Guillemdb) [reply requirements.txt]: moving back to 2.12.7 then

**Comment by Guillemdb** [comment]
@vmarkovtsev As there is no code yet the CI won't pass. I can choose to screw up the CI files to make the CI checks pass, or I can wait for the next PR to include the code.

**Comment by zurk** [comment]
you can add `@unittest.skip` decorator for now. I think it is fine

**Comment by zurk** [comment]
@Guillemdb CI is failed. Please request a review when you fix it.

**Comment by Guillemdb** [comment]
@zurk I have no idea why the build fails. It is obviously missing some files, but I don't know what are the files that need to be there so the CI passes. 

[PR#4](https://github.com/src-d/ml-core/pull/4) is working like a charm. Is really necessary that I spend more time trying to fix something that works there, and then rebasing everything?

File: Dockerfile (hunk 2 above)
Comment (zurk) [reply Dockerfile]: @Guillemdb did you check?

File: requirements.txt (hunk 5 above)
Comment (zurk) [reply requirements.txt]: @Guillemdb did you check?

File: Dockerfile (hunk 2 above)
Comment (zurk) [reply Dockerfile]: ok, we use the old client so it is the same requirements.

File: requirements.txt (hunk 5 above)
Comment (zurk) [reply requirements.txt]: ok, modelforge does not require tqdm

**Comment by zurk** [comment]
Vadim's feedback is fulfilled so I am merging

**Pull request merged**